Add exploring-p2p-hashrate-markets-v2 review to bitcoinplusplus/layer-2 (#414)
bitcointranscripts/bitcointranscripts: bitcoinplusplus/layer-2/exploring-p2p-hashrate-markets-v2.md

And try and get paid for these shares or try and submit shares that are under the target.

## Understanding Mining Pools

Cool, so mining pools.
I'm not gonna go through this entire graph, but this is in essence how a mining pool is put together, and this will be on Revlit, so if you guys do want to refer to it, you can.

*[Audience]: Did you name that Kafka, or is that actually the name of some protocol?*

Oh, that's the name of, yeah, it's like a piece of software.
No, I didn't name it.
Some mining pools use Kafka, some others, it's arbitrary.
But the essence of a mining pool is that you have a node that's connected to the network, it's scraping for those transactions, those transactions get put together then into a block, and that, well, it gets put together into the branches of the Merkle tree that we just described that then gets fed into a job Through the Stratum server to the miners.
And then the miners work on that until they have, they find a valid share and then hopefully one of those shares will be below, or one of those hashes will be below the network's difficulty.
And then that's what gets pushed back to the nodes and gossip throughout the network.

> ## Bitcoin Transactions and Verification

Reviewer: I appreciate you adding chapter titles to enhance the transcript's structure and navigability. While the effort is much appreciated, I wonder if we might find a more fitting title for this section. Often, referring to the speaker's slides can provide direct insights into suitable chapter titles. For this particular section, considering the content, perhaps "Pool Payout Formats" might be a more descriptive title?

Additionally, based on the speaker's slides, it seems there may be opportunities to introduce a few more chapter breaks leading up to the "P2P Hashrates" chapter you've already added. Adding these would further improve the readability and organization of the transcript, making it easier for readers to navigate through the content.

And yeah, it creates this situation, right, where nobody's really out trying to buy these future hash rate contracts in the traditional financial markets.
Other considerations are, well, what if you want to actually take delivery?
There's nothing really like that.
What about the duration of this contract?
So for the most part, if you look at marketplaces like NiceHash and Mining Regret tools, you have pretty short contract durations there, which makes it okay if you're kind of trying to do the betting in one direction or another, but it's fairly short.
So you're not really buying something and then just kind of letting it sit there and do its thing.
And then in terms of the actual non-deliverables or like the swaps, like contracts, well, it's, you know, you're buying kind of like a traditional financial instrument and it's not code.

> Oh, wow, okay.
> That was fast.
> Well, I'm gonna run through this really quick.
> Thank you.

Reviewer: It seems these lines are the speaker's reaction to being informed that their time is almost up. Without additional context, this moment might not be fully understood by someone reading the transcript. Providing a brief explanatory note could help clarify this situation for readers, ensuring they grasp the context of the speaker's sudden shift in pace. For example:
```suggestion
[Note: Speaker reacts to being informed they are almost out of time.]: "Oh, wow, okay.
That was fast.
Well, I'm gonna run through this really quick.
Thank you."
```

## P2P Hashrates

So our model is where you go, we have auctions set up in order to have price discovery.
We have multi-sig where the payment to the seller of the hash rate, or the miner, only occurs after the hash rate's been delivered, and we deliver the hash rate through a proxy.
So we're able to see exactly how much hash rate ends up getting delivered.
So Here we have a little model of how this works.
So we create this little agreement here in the middle.
The agreement, the purchase part here in orange at the bottom is held on chain in multisig.

> The delivery for The terms of delivery are held through the pool account.

Reviewer: Another example of false-start:
```suggestion
The terms of delivery are held through the pool account.
```

So the hash rate gets pointed here from the miner to a pool where eventually a coinbase hits the pool and the payout goes to a mining pool account.
And that's where this part of the agreement comes into play.

> And finally, you can actually obviously remove your funds from your account model to the next model.
> So you have your hot dog shape.

Reviewer: It seems there might be some uncertainty about the phrase "hot dog shape" in this part of the transcript. If the audio isn't clear enough to confirm what the speaker is saying, it's perfectly acceptable to use "[inaudible]" to indicate segments that are difficult to hear or understand. This ensures our transcript remains accurate and reliable, even when facing audio challeng
```suggestion
And finally, you can actually obviously remove your funds from your account model to the next model, so you have your [inaudible] shape.
```

And the way in which we actually are able to ensure that that delivery is happening is that we have a proxy that we can read that the hash rate is getting delivered to that pool account.
So we have a demo running right now, which you can sign up using this QR code, which I'll give another link to in a second.
This is the basic flow of what's happening with our model, but basically, we have an event-driven system where as soon as you make a payment to us, we'll switch a hash rate that is currently pointed elsewhere on our proxy straight to your guys' pool account, the pool account that you have registered with us and our database.

## Lightning 

Last notes on lightning, we want to be able to bring mining pool payouts over lightning.
The issue with that is that, I mean this is, I don't know if you guys saw the Jade, basically it produces hashes.
And the idea here is like, we have 10 sats per terahash per second per hour.
How would you be able to stream that over lightning?
Would that be possible?

> Could we use every spare or wasted jewel and convert that into hashes or gigahashes or whatever it may be?

Reviewer: Not "jewel" but "joule" the unit of energy
```suggestion
Could we use every spare or wasted joule and convert that into hashes or gigahashes or whatever it may be?
```

> You're going to exhaust those channel capacities immediately.

Reviewer: I can't comment directly on the line.
This comment is for the above line "The problem is, as Lisa calls it, we have terabat liquidity issues with mining pool payouts to miners over lightning, Because we're only pushing liquidity in one direction."

It appears there's a slight transcription error regarding the term used by the speaker. Based on the slides, the correct term is "terribad," not "terabat." Although it's a minor detail, ensuring the accuracy of such terms is crucial for maintaining the fidelity of the transcript to what was actually said.
